File: subarray_divisions/sd.js
const sum = (nums) => {
    let sum = 0;
    console.log(nums);
    for (let i = 0; i < nums.length; i++) {
        sum += nums[i];
    }
    console.log(sum);
    return sum;
}


const birthday = (s, d, m) => {
    let count = 0;
    for (let i = 0; i < s.length; i++) {
        if(sum(s.slice(i, i + m)) === d) {
            count++;
        }
    }
    return count;
}

console.log(birthday([4, 2, 1, 3, 5], 4, 2));

File: subarray_divisions/sd.py
def birthday(s:list, d:int, m:int):
    """
    >>> birthday([4, 2, 1, 3, 5], 4, 2)
    3
    >>> birthday([1, 1, 1, 1, 1, 1], 3, 5)
    0
    >>> birthday([4], 4, 1)
    1
    >>> birthday([4], 4, 0)
    0
    >>> birthday([], 4, 0)
    0
    """
    count = 0
    for i in range(len(s)):
        if sum(s[i:i+m]) == d:
            count += 1
    return count


if __name__ == "__main__":
    print(birthday([4, 2, 1, 3, 5], 4, 2))